proto/workflow_execution_service.proto

```diff
@@ -0,0 +1,237 @@
+// default settings.
+message service_info {
+  // A map with keys as the workflow format type name (currently only CWL and WDL are used
+  // although a service may support others) and value is a workflow_type_version object which
+  // simply contains an array of one or more version strings
+  map<string, workflow_type_version> workflow_type_versions = 1;
+
+  // The version(s) of the WES schema supported by this service
+  repeated string supported_wes_versions = 2;
+
+  // The filesystem protocols supported by this service, currently these may include common
+  // protocols such as 'http', 'https', 'sftp', 's3', 'gs', 'file', 'synapse', or others as
+  // supported by this service.
+  repeated string supported_filesystem_protocols = 3;
+
+  //The engine(s) used by this WES service, key is engine name e.g. Cromwell and value is version
+  map<string,string> engine_versions = 4;
+
+  //The system statistics, key is the statistic, value is the count of workflows in that state.
+  // See the State enum for the possible keys.
+  map<string,uint32> system_state_counts = 5;
+
+  // A key-value map of arbitrary, extended metadata outside the scope of the above but useful
+  // to report back
+  map<string,string> key_values = 6;
+}
+
+// Small description of workflows, returned by server during listing
+message workflow_desc {
+  // REQUIRED
+  string workflow_id  = 1;
+
+  // REQUIRED
+  state state = 2;
+}
+
+// Available workflow types supported by a given instance of the service.
+message workflow_type_version {
+    // an array of one or more version strings
+    repeated string workflow_type_version = 1;
+}
+
+message workflow_run_id {
+  // workflow ID
+  string workflow_id = 1;
+}
+
+message workflow_status {
+  // workflow ID
+  string workflow_id = 1;
+  // state
+  state state = 2;
+}
+
+message workflow_log {
+  // workflow ID
+  string workflow_id = 1;
+
+  // The original request message used to initiate this execution.
+  workflow_request request = 2;
+
+  // state
+  state state = 3;
+
+  // the logs, and other key info like timing and exit code, for the overall run of this workflow
+  log workflow_log = 4;
+
+  // the logs, and other key info like timing and exit code, for each step in the workflow
+  repeated log task_logs = 5;
+
+  // the outputs
+  google.protobuf.Struct outputs = 6;
+}
+
+// Log and other info
+message log {
+  // The task or workflow name
+  string name = 1;
+  // The command line that was run
+  repeated string cmd = 2;
+  // When the command was executed
+  string startTime = 3;
+  // When the command completed
+  string endTime = 4;
+  // Sample of stdout (not guaranteed to be entire log)
+  string stdout = 5;
+  // Sample of stderr (not guaranteed to be entire log)
+  string stderr = 6;
+  // Exit code of the program
+  int32  exitCode = 7;
+}
+
+// Enumeration of states for a given workflow request
+enum state {
```

Comment:
This is basically just recasting the TES state declaration ( https://github.com/ga4gh/task-execution-schemas/blob/master/task_execution.proto#L355 ) but with a different order for the values ( Initializing = 8; vs  INITIALIZING = 2; ) and the string equivalents will be 'Initializing' vs 'INITIALIZING'.